### PR TITLE
Feat: Added Support for CloudTrace Resource

### PR DESCRIPTION
**Pull request by aryanagrawal22**
## Description
Issue #247  Added support for CloudTrace resource 

## Changes Made
- Created client factory for CloudTrace resource : cloud_trace_client.py
- Created a new crawler factory in the crawler package : cloud_trace_crawler.py
- Also modified example.config
- In the scanner.py file, included the crawler in the crawl loop and updated the crawl_client_map. 
- Added unit tests in test_unit.py
- Left a blank cloud_traces file for test output check

## Testing
Tested the resource locally. Ran Successfully.

### CODE REVIEW

File: test/cloud_traces [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (mshudrak) [inline test/cloud_traces:1]: Before diving into the review. Could you please provide an example output here for testing?

Comment (aryanagrawal22) [reply test/cloud_traces]: Sure, I have added tests and made a minor change to an error. Please let me know any further changes.

File: test/bootstrap/cloudtrace.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# Enable the CloudTrace API
+gcloud services enable cloudtrace.googleapis.com
```
Comment (mshudrak) [inline test/bootstrap/cloudtrace.sh:4]: Well, it is not enough to generate a good test, right? You need to create this trace too to have it in the output of the tool. Could you provide an example on how to do it via bash?

File: example_config [hunk 3]
```diff
@@ -55,6 +55,9 @@
   "cloud_functions": {
     "fetch": true
   },
+  "cloud_traces":{
```
Comment (mshudrak) [inline example_config:58]: NIT: missing space

**Review by mshudrak — CHANGES_REQUESTED** [review]

**Comment by mshudrak** [comment]
Closing this one due to inactivity.

**Pull request closed without merging**